lib/dialects/postgres/query-generator.js

```diff
@@ -800,7 +800,7 @@ class PostgresQueryGenerator extends AbstractQueryGenerator {
       values = dataType.toString().match(/^ENUM\(.+\)/)[0];
     }
 
-    let sql = `CREATE TYPE ${enumName} AS ${values};`;
+    let sql = `DO $$ BEGIN IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'enum_${ tableName }_${ attr }') THEN CREATE TYPE ${ enumName } AS ${ values }; END IF; END$$;`;
```

Comment:
`typname = 'enum_${ tableName }_${ attr }'` it should check for `enumName`, like this `typname='${enumName}'`

Reply:
This solution will not work with schemas, try this https://stackoverflow.com/a/48382296

Reply:
> This solution will not work with schemas, try this https://stackoverflow.com/a/48382296

I don't know anything about schemas, and I don't have time to study them right now.